src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/Switch/SwitchHandler.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Xamarin.Platform.Handlers
+{
+	public partial class SwitchHandler
+	{
+		public static PropertyMapper<ISwitch, SwitchHandler> SwitchMapper = new PropertyMapper<ISwitch, SwitchHandler>(ViewHandler.ViewMapper)
+		{
+			[nameof(ISwitch.IsToggled)] = MapIsToggled,
+			[nameof(ISwitch.OnColor)] = MapOnColor,
+			[nameof(ISwitch.ThumbColor)] = MapThumbColor
+		};
+
+		public static void MapIsToggled(SwitchHandler handler, ISwitch view) => handler.TypedNativeView?.UpdateIsToggled(view);
```

Comment:
We can have all the logic of the Handler in a single class. However, this will require quite a few compilation directives:
- Namespaces.
- CreateView.
- Teardown.
- GetDesiredSize.
- Specific native control events, etc.

This makes reading the code more complicated.
For that reason, I think a simple way to maintain everything is:
- Map all the properties in the common handler class.
- Use extension methods to do the necessary logic.
- Have the specific platform code in .Android, .iOS, etc. classes.

Have sense?

Reply:
Reviewed and approved in a team meeting.